microsoft/testsuites/core/serial_console.py

```diff
@@ -0,0 +1,38 @@
+from lisa import (
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.features import SerialConsole
+from lisa.sut_orchestrator.azure.features import SerialConsole as AzureSerialConsole
+
+
+@TestSuiteMetadata(
+    area="serial_console",
+    category="functional",
+    description="""
+    This tests functionality of connecting to serial console.
+    """,
+)
+class SerialConsoleSuite(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        The test runs `ls` command on serial console and verifies
+        that the command has been successfully put to the
+        serial console.
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            supported_features=[SerialConsole],
+        ),
+    )
+    def verify_serial_console(self, log: Logger, node: Node) -> None:
+        command = "ls"
+        serial_console = node.features[SerialConsole]
+        assert isinstance(serial_console, AzureSerialConsole)
+        output = serial_console.run(command)
```

Comment:
Does it actually run command? I feel it's not easy, because it needs to login firstly.

Reply:
This essentially writes the command to the serial console and we just check if that it is present in the serial console output

Reply:
The input and output is not 1:1 mapping. We should assume the web socket stable at least when it's assigned. We can control underlying when to deliver a new connection or reuse existing one.

Reply:
It's just to write and echo back. If it uses "ls", it's confusing, people thought a command run. If it's really needed, use a mockup string like "echo back", so it can self explained.